spring-cloud-stream/src/main/java/org/springframework/cloud/stream/provisioning/ProvisioningProvider.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.provisioning;
+
+import org.springframework.cloud.stream.binder.ConsumerProperties;
+import org.springframework.cloud.stream.binder.ProducerProperties;
+
+/**
+ * Provisioning SPI that allows the users to provision destinations such as queues and topics.
+ * This SPI will allow the binders to be separated from any provisioning concerns and only focus
+ * on setting up endpoints for sending/receiving messages.
+ *
+ * Implementations must implement the following methods:
+ *
+ * <ul>
+ * <li>{@link #provisionProducerDestination(String, ProducerProperties)}</li>
+ * <li>{@link #provisionConsumerDestination(String, String, ConsumerProperties)} </li>
+ * </ul>
+ *
+ * @param <C>  the consumer properties type
+ * @param <P>  the producer properties type
+ *
+ * @author Soby Chacko
+ *
+ * @since 1.2
+ */
+public interface ProvisioningProvider<C extends ConsumerProperties, P extends ProducerProperties> {
+
+	/**
+	 * Creates target destinations for outbound channels. The implementation
```

Comment:
Typically the methods JavaDocs is in the imperative.
I guess mentioning of the `outbound channels` is overhead here.
Might be something more generic, like " outbound bind entities".